otelslog: set error field as record.SetErr (#8774)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

- Set error field as `record.SetErr` instead of a plain attribute in `go.opentelemetry.io/contrib/bridges/otellogrus`. (#8776)
- Set the "error" field (e.g. created via `zap.Error`) as `record.SetErr` instead of a plain attribute in `go.opentelemetry.io/contrib/bridges/otelzap`. (#8719)

> - Convert `error`-typed attributes to semantic convention error fields in `go.opentelemetry.io/contrib/bridges/otelslog`. (#8746)

Reviewer: Can you update the changelog entry to follow the pattern used in similar PRs?

- Set error field as `record.SetErr` instead of a plain attribute in `go.opentelemetry.io/contrib/bridges/otellogrus`. (#8776)
- Set the "error" field (e.g. created via `zap.Error`) as `record.SetErr` instead of a plain attribute in `go.opentelemetry.io/contrib/bridges/otelzap`. (#8719)

> - Set the "error" field (e.g. from `slog.Any("error", err)`) as `record.SetErr` instead of a plain attribute in `go.opentelemetry.io/contrib/bridges/otelslog`. (#8746)

Reviewer: Maybe

```suggestion
- Set fields implementing `error` interface from `slog` records as `record.SetErr` instead of plain attributes in `go.opentelemetry.io/contrib/bridges/otelslog`. (#8746)
```
